src/dpdk_statefulset_updater.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module used to update Kubernetes Statefulset to support DPDK."""
+
+import logging
+from typing import Iterable, Optional
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.core_v1 import Container
+from lightkube.resources.apps_v1 import StatefulSet
+
+logger = logging.getLogger(__name__)
```

Comment:
We set this here but never use it. We should probably log after we completed some actions, example after the `replace` command.

Reply:
True, added.